### PR TITLE
sig-release: Add/update jobs for building debs/rpms

### PR DESCRIPTION
**Pull request by justaugustus**
Dependent on https://github.com/kubernetes/release/pull/818

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

(for discussion)
/hold
/cc @kubernetes/release-engineering @kubernetes/sig-release-admins

### CODE REVIEW

**Comment by spiffxp** [comment]
/approve
needs rebase

**Comment by justaugustus** [comment]
Thanks @spiffxp!

**Comment by justaugustus** [comment]
Dang yamllint:
```
config/jobs/kubernetes-sigs/sig-storage-local-static-provisioner/sig-storage-local-static-provisioner-trusted.yaml
  40:3      warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes-sigs/kind/kind.yaml
  49:1      warning  comment not indented like content  (comments-indentation)
  101:1     warning  comment not indented like content  (comments-indentation)
  143:1     warning  comment not indented like content  (comments-indentation)
  198:1     warning  comment not indented like content  (comments-indentation)
  235:1     warning  comment not indented like content  (comments-indentation)
  272:1     warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes-sigs/cluster-api-provider-vsphere/cluster-api-provider-vsphere-ci.yaml
  22:9      warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/gcp-compute-persistent-disk-csi-driver-postsubmits.yaml
  176:11    warning  comment not indented like content  (comments-indentation)
  179:11    warning  comment not indented like content  (comments-indentation)
  212:11    warning  comment not indented like content  (comments-indentation)
  215:11    warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/sig-cli/sig-cli-config.yaml
  140:1     warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/release/release-config.yaml
  75:1      warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml
  35:1      warning  comment not indented like content  (comments-indentation)
  360:1     warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/sig-cluster-lifecycle/k8s-upgrade-gke.yaml
  30:1      warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/sig-cluster-lifecycle/k8s-upgrade-gce.yaml
  483:1     warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/sig-release/kubernetes-builds.yaml
  115:3     error    duplication of key "annotations" in mapping  (key-duplicates)
  145:9     warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/sig-aws/eks/eks-periodics.yaml
  27:1      warning  comment not indented like content  (comments-indentation)
  55:1      warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/cloud-provider-azure/cloud-provider-azure-config.yaml
  48:3      warning  comment not indented like content  (comments-indentation)
  101:3     warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/sig-testing/coverage.yaml
  60:1      warning  comment not indented like content  (comments-indentation)
 config/jobs/kubernetes/sig-testing/conformance-e2e.yaml
  30:13     warning  comment not indented like content  (comments-indentation)
  35:11     warning  comment not indented like content  (comments-indentation)
  40:15     warning  comment not indented like content  (comments-indentation)
  43:15     warning  comment not indented like content  (comments-indentation)
  47:3      warning  comment not indented like content  (comments-indentation)
  68:13     warning  comment not indented like content  (comments-indentation)
  73:11     warning  comment not indented like content  (comments-indentation)
  78:15     warning  comment not indented like content  (comments-indentation)
  81:15     warning  comment not indented like content  (comments-indentation)
  109:13    warning  comment not indented like content  (comments-indentation)
  114:11    warning  comment not indented like content  (comments-indentation)
  119:15    warning  comment not indented like content  (comments-indentation)
  122:15    warning  comment not indented like content  (comments-indentation)
```

**Comment by spiffxp** [comment]
/retest
it's not failing across all PRs, even though I see this output on green runs, what gives

File: config/jobs/kubernetes/sig-release/kubernetes-builds.yaml [hunk 1]
```diff
@@ -59,10 +59,41 @@ periodics:
...
+    testgrid-dashboards: sig-release-master-informing
+    testgrid-tab-name: build-packages-debs
+  spec:
+    containers:
+    - image: gcr.io/k8s-testimages/bootstrap:v20190703-1f4d616
+      args:
+      - --repo=k8s.io/release
+      - --root=/go/src
+      - --timeout=180
+      - --scenario=execute
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-release/kubernetes-builds.yaml]: The use of bootstrap is super deprecated, what is it about this image you require?

Check out the `periodic` example here for what to use instead (the `extra_refs` thing)

https://github.com/kubernetes/test-infra/tree/master/config/jobs#job-examples

File: config/jobs/kubernetes/sig-release/kubernetes-builds.yaml [hunk 2]
```diff
@@ -59,10 +59,41 @@ periodics:
...
+      resources:
+        requests:
+          cpu: 4
+          memory: "8Gi"
+- interval: 4h
+  name: ci-release-build-packages-rpms
+  labels:
+    preset-service-account: "true"
+    preset-dind-enabled: "true"
+  annotations:
```
Comment (alvaroaleman) [inline config/jobs/kubernetes/sig-release/kubernetes-builds.yaml]: The job already has an `annotations` key further down, thats why the yamllint fails.

**Comment by justaugustus** [comment]
@alvaroaleman -- I think I opened the PR before the annotations got applied across all job configs, so they got doubly applied. Thanks for catching that! 

@spiffxp -- thanks for the tip! I fixed it up. :)
(I think.)

File: config/jobs/kubernetes/sig-release/kubernetes-builds.yaml [hunk 3]
```diff
@@ -52,27 +52,30 @@ periodics:
...
     containers:
-    - image: gcr.io/k8s-testimages/bootstrap:v20190703-1f4d616
-      args:
-      - --repo=k8s.io/release
-      - --root=/go/src
-      - --timeout=180
-      - --scenario=execute
-      - --
-      - ./debian/jenkins.sh
+    - image: quay.io/k8s/release-tools:latest
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-release/kubernetes-builds.yaml]: Not sure what this is built from, but FWIW I would recommend a followup PR to add a `release-tools` dir to https://github.com/kubernetes/test-infra/tree/master/images and go back to using gcr.io

**Comment by justaugustus** [comment]
Thanks @spiffxp!
/hold cancel

File: config/jobs/kubernetes/sig-release/kubernetes-builds.yaml (hunk 3 above)
Comment (justaugustus) [reply config/jobs/kubernetes/sig-release/kubernetes-builds.yaml]: @spiffxp -- I was planning on adding them to k/release. Is there an advantage to having them in k/test-infra instead?

Comment (spiffxp) [reply config/jobs/kubernetes/sig-release/kubernetes-builds.yaml]: we auto-build and push images that live in that dir to gcr.io via https://github.com/kubernetes/test-infra/tree/master/images/builder, tagged with the date and sha they were built, it's a pattern that's worked pretty well to know exactly what's being used

**Pull request merged**